internal/kafka/internal/services/cloud_providers_test.go

```diff
@@ -501,3 +555,116 @@ func Test_ListCloudProviderRegions(t *testing.T) {
+	}{
+		{
+			name: "successful get cloud provider regions from various provider types",
+			fields: fields{
+				connectionFactory: db.NewMockConnectionFactory(nil),
+				providerFactory: &clusters.ProviderFactoryMock{
+					GetProviderFunc: func(providerType api.ClusterProviderType) (clusters.Provider, error) {
+						return &clusters.ProviderMock{
+							GetCloudProvidersFunc: func() (*types.CloudProviderInfoList, error) {
+								return &types.CloudProviderInfoList{
+									Items: []types.CloudProviderInfo{
+										{
+											ID:          "azure",
+											Name:        "azure",
+											DisplayName: "azure",
+										},
+										{
+											ID:          "aws",
+											Name:        "aws",
+											DisplayName: "aws",
+										},
+									},
+								}, nil
+							},
+							GetCloudProviderRegionsFunc: func(providerInf types.CloudProviderInfo) (*types.CloudProviderRegionInfoList, error) {
+								return &types.CloudProviderRegionInfoList{
+									Items: []types.CloudProviderRegionInfo{
+										{
+											ID:              "af-east-1",
+											CloudProviderID: providerInf.ID,
+											Name:            "af-east-1",
+											DisplayName:     "af-east-1",
+										},
+										{
+											ID:              "af-south-1",
+											CloudProviderID: providerInf.ID,
+											Name:            "af-south-1",
+											DisplayName:     "af-south-1",
+										},
+									},
+								}, nil
+							},
+						}, nil
+					}},
+				cache: cache.New(5*time.Minute, 10*time.Minute),
+			},
+			setupFn: func() {
+				mocket.Catcher.Reset()
+				mocket.Catcher.NewMock().
+					WithQuery("SELECT DISTINCT").
+					WithReply([]map[string]interface{}{{"provider_type": "ocm"}, {"provider_type": "standalone"}})
+					mocket.Catcher.NewMock().WithExecException().WithQueryException()
+			},
+			args: args{
+				id: "azure",
+			},
+			want: []api.CloudRegion{
+				{
+					Kind:                   "",
+					Id:                     "af-east-1",
+					DisplayName:            "af-east-1",
+					CloudProvider:          "azure",
+					Enabled:                false,
+					SupportedInstanceTypes: nil,
+					Capacity:               nil,
+				},
+				{
+					Kind:                   "",
+					Id:                     "af-south-1",
+					DisplayName:            "af-south-1",
+					CloudProvider:          "azure",
+					Enabled:                false,
+					SupportedInstanceTypes: nil,
+					Capacity:               nil,
+				},
+			},
+			wantErr: nil,
+		},
+	}
+	RegisterTestingT(t)
+	for _, tt := range tests {
+		tt.setupFn()
+		t.Run(tt.name, func(t *testing.T) {
+			p := cloudProvidersService{
+				providerFactory:   tt.fields.providerFactory,
+				connectionFactory: tt.fields.connectionFactory,
+				cache:             tt.fields.cache,
+			}
+			got, err := p.ListCachedCloudProviderRegions(tt.args.id)
+			Expect(got).To(Equal(tt.want))
+			if err != nil{
+				Expect(err).To(Equal(tt.wantErr))
+			}
```

Comment:
I think this can be reduced to 
```go
Expect(err).To(Equal(tt.wantErr))
```
given that when `tt.want == false` watErr will be `nil`